client/components/LearnerViewDirective.js

```diff
@@ -21,104 +21,174 @@ tie.directive('learnerView', [function() {
     restrict: 'E',
     scope: {},
     template: `
-      <h3>Exercise: {{title}}</h3>
-
-      <div class="tie-learner-view-left-column">
-        <div class="tie-previous-instructions">
-          <div ng-repeat="previousInstruction in previousInstructions track by $index">
-            <p ng-repeat="paragraph in previousInstruction track by $index">
-              {{paragraph}}
-            </p>
-            <hr>
+      <div id="tie-exercise-ui-outer" class="tie-exercise-ui-outer">
+        <div id="tie-exercise-ui-inner" class="tie-exercise-ui-inner">
+          <div id="tie-question-ui" class="tie-question-ui">
+            <div id="tie-question-window" class="tie-question-window">
+              <h3>{{title}}</h3>
+              <div class="tie-previous-instructions">
+                <div ng-repeat="previousInstruction in previousInstructions track by $index">
+                  <p ng-repeat="paragraph in previousInstruction track by $index">
+                    {{paragraph}}
+                  </p>
+                  <hr>
+                </div>
+              </div>
+              <div id="tie-instructions" class="tie-instructions">
+                <p ng-repeat="paragraph in instructions">
+                  {{paragraph}}
+                </p>
+              </div>
+            </div>
+            <button ng-click="showNextPrompt()" ng-if="nextButtonIsShown"
+                id="tie-next-button" class="tie-next-button">Next</button>
+            </button>
+          </div>
+          <div id="tie-coding-ui" class="tie-coding-ui">
+            <div id="tie-feedback-window" class="tie-feedback-window">
+              <div id="tie-feedback" class="tie-feedback">
+                <p ng-repeat="paragraph in feedbackMessages track by $index"
+                    class="tie-feedback-paragraph">
+                  {{paragraph}}
+                </p>
+              </div>
+            </div>
+            <div id="tie-coding-window" class="tie-coding-window">
+              <div id="tie-lang-terminal" class="tie-lang-terminal">
+                <div id="tie-coding-terminal" class="tie-coding-terminal">
+                  <ui-codemirror ui-codemirror="codeMirrorOptions" ng-model="code" class="tie-codemirror-container"></ui-codemirror>
+                </div>
+                <select id="tie-lang-select-menu" class="tie-lang-select-menu"
+                    name="lang-select-menu">
+                  <option value="Python" selected>Python</option>
+                  <option value="JavaScript">JavaScript</option>
```

Comment:
How far out are we to adding new languages? I can imagine people asking if it will support more languages than Python, and as a prototype, it shows this planned functionality and how users can change the language.